Patch-Source: https://www.kylheku.com/cgit/txr/patch/?id=d9a18c7440ac4818de7561212e041a85d4506a72
--
From d9a18c7440ac4818de7561212e041a85d4506a72 Mon Sep 17 00:00:00 2001
From: Kaz Kylheku <kaz@kylheku.com>
Date: Mon, 16 Dec 2024 17:12:50 -0800
Subject: listener: regression: txr bails when .txr-profile missing.

* parser.c (load_rcfile): When neither ~/.txr-profile nor ~/.txr_profile
exist, then bail. Do not pass nil to abs-path-p and other functions.
---
 parser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/parser.c b/parser.c
index e60166cc..ba236e65 100644
--- a/parser.c
+++ b/parser.c
@@ -1040,6 +1040,9 @@ static void load_rcfile(val home, val pexist_s, val psafe_s, val ppriv_s)
                  if2(funcall1(pexist_s, try2), try2));
   val resolved_name = name;
 
+  if (!name)
+    return;
+
   if (!funcall1(psafe_s, name)) {
     report_path_perm_problem(name);
     return;
-- 
cgit v1.2.3

